#pragma once

#include <ui/dialogs/common/button_box_dialog.h>

#include <ui/workbench/workbench_context_aware.h>

namespace Ui
{
    class ChangeUserPasswordDialog;
}

class QnChangeUserPasswordDialog: public QnButtonBoxDialog, public QnWorkbenchContextAware
{
    Q_OBJECT

    typedef QnButtonBoxDialog base_type;

public:
    QnChangeUserPasswordDialog(QWidget* parent = nullptr);
    virtual ~QnChangeUserPasswordDialog();

    virtual void done(int r) override;

    QString newPassword() const;

private:
    bool validate();

private:
    Q_DISABLE_COPY(QnChangeUserPasswordDialog)
    QScopedPointer<Ui::ChangeUserPasswordDialog> ui;
};


